### ISSUE TITLE
Adding custom loss function to discrimator  

### ISSUE THREAD

**Issue by xhuvom** [junyanz/pytorch-CycleGAN-and-pix2pix#521]
I am trying to add a custom loss function which takes real and fake images to perform a custom parametric matching. I have found the **D_loss** on 

> pix2pix_model.py 

script as
```
 # Combined loss
  self.loss_D = (self.loss_D_fake + self.loss_D_real) * 0.5
```
But however, my custom loss function will take the real and fake images as arguments and return a score. I am confused how to access the real and fake images that is taken on account in the equation above. I have found a line where the images are found on the same script as
```
 # specify the images you want to save/display. The program will call base_model.get_current_visuals
  self.visual_names = ['real_A', 'fake_B', 'real_B']
```

If the images could be accessed as numpy array, it would be feasible. Looking forward for suggestions.

**Comment by junyanz** [comment]
You can just use `self.real_A`, `self.fake_B`, `self.real_B` directly. They are PyTorch tensor arrays.
